### PR TITLE
modify update_attributes method in subscription class

### PR DESCRIPTION
**Pull request by rloomba**
This PR addresses the bug in #241.

### CODE REVIEW

File: spec/recurly/subscription_spec.rb [hunk 1]
```diff
@@ -351,4 +351,38 @@
...
   end
+
+  describe '#update_attributes' do
+    describe 'when the plan code does change' do
+      it 'sends all updated atributes to the server' do
+        stub_api_request :get, 'subscriptions/abcdef1234567890', 'subscriptions/show-200'
+
+        subscription = Subscription.find 'abcdef1234567890'
+
+        stub_request(:put, "https://api_key:@api.recurly.com/v2/subscriptions/abcdef1234567890").
```
Comment (rloomba) [inline spec/recurly/subscription_spec.rb]: I'm happy to refactor this into it's own helper method, let me know.

**Comment by bhelx** [comment]
@rloomba thanks a lot for following up so quick!

Going to review this the first thing this morning.

File: spec/recurly/subscription_spec.rb (hunk 1 above)
Comment (bhelx) [inline spec/recurly/subscription_spec.rb]: Yeah I did the same thing when I tested it. I modified the `stub_api_request` method to handle it but it was ugly. I think this is good for now. I'm going to do some refactoring of the tests soon.

File: lib/recurly/subscription.rb [hunk 2]
```diff
@@ -210,8 +210,30 @@ def update_notes(notes)
       true
     end
 
+    # Overrides the behavior of `update_attributes` in Resource class so ensure
+    # all attributes are marked as dirty if the plan code changes
```
Comment (bhelx) [inline lib/recurly/subscription.rb]: :+1: thanks for the comment

**Comment by bhelx** [comment]
:+1: thanks for getting this out so fast

**Comment by rloomba** [comment]
Yep, no worries, glad I could help!

**Pull request merged**